### PR TITLE
 Fixes compilation for react-native-windows.

### PR DESCRIPTION
**Pull request by rozele**
Updates the following for compatibility with the latest react-native-windows:
- NuGet dependencies
- *.vcxproj and *.csproj files
- Adds DLL exports for APIs called with P/Invoke from C# project

### CODE REVIEW

File: yoga/Yoga.cpp [hunk 1]
```diff
@@ -39,24 +39,24 @@ static int YGDefaultLog(const YGConfigRef config,
 #endif
 
 static YGConfig gYGConfigDefaults = {
-    .experimentalFeatures =
+    /* .experimentalFeatures = */
```
Comment (rozele) [inline yoga/Yoga.cpp]: There's still lots of designated initializer syntax to be removed, but I at least added the YGStyle and YGLayout components.

**Comment by woehrl01** [comment]
@rozele, as my PR (#746) just landed,  I think the road is free to land this PR, and if you are still interested.

File: csharp/Facebook.Yoga/YogaNode.cs [hunk 2]
```diff
@@ -20,6 +22,7 @@
 
 namespace Facebook.Yoga
 {
+#pragma warning disable CS1591 // Missing XML comment for publicly visible type or member
```
Comment (rozele) [inline csharp/Facebook.Yoga/YogaNode.cs]: >CS1591 [](start = 24, length = 6)

Delete dupe?

File: csharp/Yoga/Yoga.Universal.vcxproj [hunk 3]
```diff
@@ -241,20 +241,31 @@
       <GenerateDebugInformation>true</GenerateDebugInformation>
     </Link>
   </ItemDefinitionGroup>
-  <ItemGroup>
+    <ClInclude Include="..\..\yoga\Utils.h" />
+    <ClInclude Include="..\..\yoga\YGConfig.h" />
     <ClInclude Include="..\..\yoga\YGEnums.h" />
```
Comment (rozele) [inline csharp/Yoga/Yoga.Universal.vcxproj]: Missing ItemGroup now?

File: csharp/Yoga/Yoga.Universal.vcxproj [hunk 4]
```diff
@@ -241,20 +241,31 @@
...
     <ClInclude Include="..\..\yoga\YGMacros.h" />
-    <ClInclude Include="..\..\yoga\YGNodeList.h" />
+    <ClInclude Include="..\..\yoga\YGNode.h" />
+    <ClInclude Include="..\..\yoga\YGNodePrint.h" />
+    <ClInclude Include="..\..\yoga\YGStyle.h" />
+    <ClInclude Include="..\..\yoga\Yoga-internal.h" />
+    <ClInclude Include="..\..\yoga\Yoga.h" />
     <ClInclude Include="resource.h" />
-    <ClInclude Include="YGInterop.h" />
     <ClInclude Include="stdafx.h" />
```
Comment (rozele) [inline csharp/Yoga/Yoga.Universal.vcxproj]: Why delete this?

**Comment by rozele** [comment]
Thanks @woehrl01 - I just rebased this branch off your changes. Would be great to get your feedback on this PR now that it's rebased.

**Comment by woehrl01** [comment]
Looks good to me @rozele. It would be lovely if instead of using a dedicated `#pragma`, `<DisableSpecificWarnings>1587;1591</DisableSpecificWarnings>` is used inside the `vcxproj`, so that the code isn't adapted specific to MSVC. What do you think about that?

**Comment by NickGerleman** [comment]
@rozele I'm going to hazard to guess this PR isn't going to be actioned on. Going to close this, but feel free to reopen.

**Pull request closed without merging**